server/src/test/java/org/opensearch/ratelimitting/admissioncontrol/AdmissionControlSingleNodeTests.java

```diff
@@ -120,20 +222,86 @@ public void testAdmissionControlRejectionMonitorOnlyMode() throws Exception {
+                .getRejectionCount()
+                .get(AdmissionControlActionType.SEARCH.getType())
+        );
+
+        updateSettingsRequest = new ClusterUpdateSettingsRequest();
+        updateSettingsRequest.transientSettings(
+            Settings.builder()
+                .put(super.nodeSettings())
+                .put(SEARCH_IO_USAGE_LIMIT.getKey(), 0)
+                .put(INDEXING_IO_USAGE_LIMIT.getKey(), 0)
+                .put(SEARCH_CPU_USAGE_LIMIT.getKey(), 101)
+                .put(INDEXING_CPU_USAGE_LIMIT.getKey(), 101)
+        );
+        assertAcked(client().admin().cluster().updateSettings(updateSettingsRequest).actionGet());
+        bulk = client().prepareBulk();
+        for (int i = 0; i < 3; i++) {
+            bulk.add(client().prepareIndex("index").setSource("foo", "bar " + i));
+        }
+        // verify bulk request success but admission control having rejections stats
+        res = client().bulk(bulk.request()).actionGet();
+        assertFalse(res.hasFailures());
+        acStats = admissionControlService.getStats();
+        assertEquals(
+            1,
+            (long) acStats.get(CpuBasedAdmissionController.CPU_BASED_ADMISSION_CONTROLLER)
+                .getRejectionCount()
+                .get(AdmissionControlActionType.INDEXING.getType())
+        );
+        if (Constants.LINUX) {
+            assertEquals(
+                1,
+                (long) acStats.get(IoBasedAdmissionController.IO_BASED_ADMISSION_CONTROLLER)
+                    .getRejectionCount()
+                    .getOrDefault(AdmissionControlActionType.INDEXING.getType(), 0L)
+            );
+        } else {
+            assertNull(acStats.get(IoBasedAdmissionController.IO_BASED_ADMISSION_CONTROLLER));
+        }
+        searchRequest = new SearchRequest("index");
+        searchResponse = client().search(searchRequest).actionGet();
+        assertEquals(3, searchResponse.getHits().getHits().length);
+        acStats = admissionControlService.getStats();
+        assertEquals(
+            1,
+            (long) acStats.get(CpuBasedAdmissionController.CPU_BASED_ADMISSION_CONTROLLER)
+                .getRejectionCount()
+                .get(AdmissionControlActionType.SEARCH.getType())
+        );
+        if (Constants.LINUX) {
+            assertEquals(
+                1,
+                (long) acStats.get(IoBasedAdmissionController.IO_BASED_ADMISSION_CONTROLLER)
+                    .getRejectionCount()
+                    .getOrDefault(AdmissionControlActionType.SEARCH.getType(), 0L)
+            );
+        } else {
+            assertNull(acStats.get(IoBasedAdmissionController.IO_BASED_ADMISSION_CONTROLLER));
+        }
     }
 
     public void testAdmissionControlRejectionDisabledMode() throws Exception {
         assertBusy(() -> assertEquals(1, getInstanceFromNode(ResourceUsageCollectorService.class).getAllNodeStatistics().size()));
+        Thread.sleep(6000);
```

Comment:
Same here - if we need to increase the wait time, then we can increase the timeout in assertbusy

Reply:
Removed.